ui/packages/catalog-ui-search/src/main/webapp/component/query-advanced/query-advanced.view.js

```diff
@@ -13,77 +13,97 @@
+        'click .editor-edit': 'edit',
+        'click .editor-cancel': 'cancel',
+        'click .editor-save': 'save'
+    },
+    regions: {
+        querySettings: '.query-settings',
+        queryAdvanced: '.query-advanced'
+    },
+    ui: {
+    },
+    initialize: function() {
+        this.$el.toggleClass('is-form-builder', this.options.isFormBuilder === true);
+        this.$el.toggleClass('is-form', this.options.isForm === true);
+    },
+    onBeforeShow: function(){
+        this.model = this.model._cloneOf ? store.getQueryById(this.model._cloneOf) : this.model;
+        this.querySettings.show(new QuerySettingsView({
+            model: this.model,
+            isForm: this.options.isForm || false,
+            isFormBuilder: this.options.isFormBuilder || false
+        }));
+        this.queryAdvanced.show(new FilterBuilderView({
+            model: new FilterBuilderModel(),
+            isForm: this.options.isForm || false,
+            isFormBuilder: this.options.isFormBuilder || false
+        }));
+
+        if (this.options.isForm === true && this.model.get('filterTree') !== undefined) {
+            this.queryAdvanced.currentView.deserialize(cql.simplify(JSON.parse(this.model.get('filterTree'))));
+        }
+        else if (this.model.get('cql')) {
+            this.queryAdvanced.currentView.deserialize(cql.simplify(cql.read(this.model.get('cql'))));
+        }
+        this.queryAdvanced.currentView.turnOffEditing();
+        this.edit();
+    },
+    focus: function(){
+        var tabbable =  _.filter(this.$el.find('[tabindex], input, button'), function(element){
+            return element.offsetParent !== null;
+        });
+        if (tabbable.length > 0){
+            $(tabbable[0]).focus();
+        }
+    },
+    edit: function(){
+        this.$el.addClass('is-editing');
+        this.querySettings.currentView.turnOnEditing();
+        this.queryAdvanced.currentView.turnOnEditing();
+        if (this.options.isForm === true && this.options.isFormBuilder !== true) {
             this.queryAdvanced.currentView.turnOffEditing();
-            this.edit();
-        },
-        focus: function(){
-            var tabbable =  _.filter(this.$el.find('[tabindex], input, button'), function(element){
-                return element.offsetParent !== null;
-            });
-            if (tabbable.length > 0){
-                $(tabbable[0]).focus();
-            }
-        },
-        edit: function(){
-            this.$el.addClass('is-editing');
-            this.querySettings.currentView.turnOnEditing();
-            this.queryAdvanced.currentView.turnOnEditing();
-        },
-        cancel: function(){
-            this.$el.removeClass('is-editing');
-            this.onBeforeShow();
-        },
-        save: function(){
-            this.$el.removeClass('is-editing');
-            this.querySettings.currentView.saveToModel();
+            //TODO: Deal with the oddities in turning off editing in that view
```

Comment:
Does this TODO still need to be addressed?

Reply:
Yeah it should be